install/docker/index.markdown

```diff
@@ -31,25 +31,46 @@ And you will likely need to log out and back into your user account for the chan
 
 ## Running MoveIt Containers
 
-To run a Debian-installed container of MoveIt with graphics support:
+To use the command line only (without GUI interfaces like Rviz) in a non-persistent container, run this command:
 
-    wget https://raw.githubusercontent.com/ros-planning/moveit/master/.docker/gui-docker -O gui-docker && chmod +x gui-docker
-    ./gui-docker -it --rm moveit/moveit:melodic-release /bin/bash
+    docker run -it moveit/moveit:master-source bash
 
-This will attempt to use nvidia-docker if hardware and drivers are available.
+To use Rviz (and other GUIs), you need to set up hardware acceleration for Docker as described [here](http://wiki.ros.org/docker/Tutorials/Hardware%20Acceleration#nvidia-docker2). First, install nvidia-docker2 according to [these instructions](https://github.com/nvidia/nvidia-docker/wiki/Installation-(version-2.0)).
 
-**Command-Line Only**
+Then, use the commands below to:  
+- create a directory 
+- download the Dockerfile and entrypoint script
+- build the Docker image locally for the Nvidia runtime
+    
+    mkdir local_moveit_image & cd local_moveit_image
+    wget https://raw.githubusercontent.com/ros-planning/moveit/master/.docker/run-gui-moveit-docker && chmod +x run-gui-moveit-docker
+    wget https://raw.githubusercontent.com/ros-planning/moveit/master/.docker/Dockerfile-nvidia
+    docker build --file Dockerfile-nvidia -t local_moveit_image .
 
-    docker run -it moveit/moveit:melodic-release
+Now, you can execute the script to create and run a persistent container:
+
+    ./run-gui-moveit-docker
+
+The container is persistent, so changes you make to your container will remain. Running `run-gui-moveit-docker` in multiple terminals will connect them to the same container. To reset the container, simply delete it using the commands below. It will be regenerated next time you run `run-gui-moveit-docker`:
+
+    docker stop local_moveit_container && docker rm local_moveit_container
+
+To rebuild the image (e.g. to get the latest changes on the master branch), run:
+
+    docker build --file Dockerfile-nvidia -t local_moveit_image .
+
+You can test that the GUI works by running rviz:
+
+    roscore > /dev/null & rosrun rviz rviz
```

Comment:
I could get the docker image running, but unfortunately rviz didn't run. Instead on this command nothing happens for a while (~30 seconds) and upon `^C`, I get:
```
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
No protocol specified
qt.qpa.screen: QXcbConnection: Could not connect to display :1
Could not connect to any X display.
```
Successive invocations don't hang and directly give the error.
Don't have much experience with Docker, is there something obvious I should have missed?